gfxr_ext/decode/dive_annotation_processor_test.cpp

```diff
@@ -0,0 +1,105 @@
+/*
+ Copyright 2025 Google LLC
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+#include "dive_annotation_processor.h"
+
+#include <gtest/gtest.h>
+
+namespace gfxrecon::decode
+{
+namespace
+{
+
+class DiveAnnotationProcessorTestFixture : public testing::Test
+{
+protected:
+    DiveAnnotationProcessor processor_;
+
+    // Helper function to compare VulkanCommandInfo
+    bool AreVulkanCommandInfoEqual(const DiveAnnotationProcessor::VulkanCommandInfo& actual,
+                                   const std::string&                                expected_name,
+                                   uint32_t                                          expected_index,
+                                   const nlohmann::ordered_json&                     expected_args)
+    {
+        return actual.GetVkCmdName() == expected_name && actual.GetVkCmdIndex() == expected_index &&
+               actual.GetArgs() == expected_args;
+    }
+
+    // Helper function to compare SubmitInfo
+    bool AreSubmitInfoEqual(const DiveAnnotationProcessor::SubmitInfo* actual,
+                            const std::string&                         expected_name,
+                            uint32_t expected_command_buffer_count)
+    {
+        return actual != nullptr && actual->GetSubmitText() == expected_name &&
+               actual->GetCommandBufferCount() == expected_command_buffer_count;
+    }
+};
+
+TEST_F(DiveAnnotationProcessorTestFixture, WriteBlockEndSingleSubmit)
+{
+    gfxrecon::util::DiveFunctionData data("vkQueueSubmit", 0, 1, { { "arg1", 1 } });
```

Comment:
Consider function argument comments to give these numbers some more meaning: https://google.github.io/styleguide/cppguide.html#Function_Argument_Comments

```suggestion
    gfxrecon::util::DiveFunctionData data("vkQueueSubmit", /*cmd_buffer_index=*/0, /*block_index=*/1, { { "arg1", 1 } });
```

Reply:
Done